src/sentry/dynamic_sampling/rules_generator.py

```diff
@@ -120,11 +103,20 @@ def generate_boost_release_rules(project_id: int, sample_rate: float) -> List[Re
                     "op": "and",
                     "inner": [
                         {
-                            "op": "glob",
+                            "op": "eq",
                             "name": "trace.release",
                             "value": [boosted_release.version],
                         },
-                        _generate_environment_condition(boosted_release.environment),
+                        {
+                            "op": "eq",
+                            "name": "trace.environment",
+                            # When environment is None, it will be mapped to equivalent null in json.
+                            # When Relay receives a rule with "value": null it will match it against events without
+                            # the environment tag set.
+                            "value": [boosted_release.environment]
+                            if boosted_release.environment
+                            else None,
```

Comment:
```suggestion
                            "value": boosted_release.environment
```

`eq` also works with a single value, it then goes into this branch:
https://github.com/getsentry/relay/blob/ae0f47aa5d80bf9086f8bef6575c3713eaf428e2/relay-sampling/src/lib.rs#L68